src/http_request.cpp

```diff
@@ -127,6 +127,10 @@ const std::map<std::string, std::string, http::arg_comparator> http_request::get
     return arguments;
 }
 
+file_info_s &http_request::get_or_create_file_info(const char *upload_file_name) {
```

Comment:
nit: we generally attach '&' and '*' to the preceding term in declarations of inputs and output (check across the whole CR).

Also, I'd prefer passing around `std::string` over `const char*`.

Can the return be marked const?

Reply:
The attached `&` and `*` are adjusted. Additionally this function takes `const std::string& upload_file_name` as parameter now.

The return value is now const.
Therefor I introduced new setters for the content of the file_info_s as these where previously set in the calling function directly on the returned reference.